salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+            ret {dict} -- salt state result
+        """
+        log.info('Cleaning up secrets engines...')
+        has_changes = False
+
+        try:
+            for secret_engine in remote_engines:
+                if not (secret_engine.type == "system" or
+                        secret_engine.type == "cubbyhole" or
+                        secret_engine.type == "identity" or
+                        secret_engine.type == "generic"):
+                    if secret_engine in local_engines:
+                        log.debug('Secrets engine "%s" at path "%s" exists in configuration, no cleanup necessary.',
+                                  secret_engine.type, secret_engine.path)
+                    else:
+                        log.debug('Secrets engine "%s" at path "%s" does not exist in configuration. Disabling...',
+                                  secret_engine.type, secret_engine.path)
+                        has_changes = True
+                        client.sys.disable_secrets_engine(
+                            path=secret_engine.path
+                        )
+                        log.debug('Secrets engine "%s" at path "%s" is disabled.',
+                                secret_engine.type, secret_engine.type)
+        except Exception:
+            raise
+
+        if has_changes:
+            ret['changes']['new'] = ["type: {} - path: {}".format(ob.type, ob.path) for ob in local_engines]
+
+        log.info('Finished cleaning up secrets engines.')
+
+
+class VaultAuditManager():
+    """
+    Module for handling Vault Audit Devices
+    """
+
+    def __init__(self):
+        """Initialize Vault Audit Managers
+        """
+        log.info("Initializing Vault Audit Manager...")
+
+    def get_remote_audit_devices(self, client, ret):
+        """Get audit devices information from remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- salt state result
+
+        Returns:
+            list -- audit devices
+        """
+        log.info("Retrieving audit devices from vault...")
+        devices = []
+        try:
+            audit_devices_resp = client.sys.list_enabled_audit_devices()
+            log.debug(audit_devices_resp)
+            for device in audit_devices_resp['data']:
+                audit_device = audit_devices_resp[device]
+                devices.append(
+                    VaultAuditDevice(
+                        type=audit_device['type'],
+                        path=(audit_device["path"]
+                              if 'path' in audit_device else device),
+                        description=audit_device["description"],
+                        config=OrderedDict(
+                            sorted(audit_device["options"].items()))
+                    )
+                )
+
+            log.info('Finished retrieving audit devices from vault.')
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.